search-solr/src/main/java/edu/unc/lib/boxc/search/solr/services/SearchStateFactory.java

```diff
@@ -397,13 +397,35 @@ private void populateSearchState(SearchState searchState, Map<String,String[]> r
                 }
             }
             searchState.setFacetsToRetrieve(facetsToRetrieve);
+
+
+            //Add date added year range, which is a sort of facet
+            try {
+                SearchState.RangePair dateCreatedYear = new SearchState.RangePair();
+                parameter = getParameter(request, searchSettings.searchFieldParam(
+                        SearchFieldKey.DATE_CREATED_YEAR.name()));
+                if (parameter != null && parameter.length() > 0) {
+                    String[] dateRange = parameter.split(",");
+                    dateCreatedYear.setLeftHand(dateRange[0]);
+                    dateCreatedYear.setRightHand(dateRange[1]);
+                }
+
+                if (dateCreatedYear.getLeftHand() != null && dateCreatedYear.getRightHand() != null) {
```

Comment:
I think the RangePair validation method would catch if both were null, and it should be fine if one is null, so can probably remove this check